cdap-unit-test/src/test/java/io/cdap/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -1036,6 +1037,54 @@ public void testGetServiceURL() throws Exception {
     centralServiceManager.waitForStopped(10, TimeUnit.SECONDS);
   }
 
+  @Category(SlowTests.class)
+  @Test
+  public void testGetServiceURLDiffNamespace() throws Exception {
+    ApplicationManager discoveringApplicationManager = deployApplication(AppUsingGetServiceURL.class);
+    ServiceManager discoveringServiceManager = discoveringApplicationManager
+      .getServiceManager(AppUsingGetServiceURL.FORWARDING).start();
+    discoveringServiceManager.waitForRun(ProgramRunStatus.RUNNING, 10, TimeUnit.SECONDS);
+
+    NamespaceId systemSpace = new NamespaceId("system");
+    ApplicationManager systemApplicationManager = deployApplication(systemSpace, AppWithServices.class);
+    ServiceManager systemCentralServiceManager =
+      systemApplicationManager.getServiceManager(AppWithServices.SERVICE_NAME).start();
+    systemCentralServiceManager.waitForRun(ProgramRunStatus.RUNNING, 10, TimeUnit.SECONDS);
+
+    String result = callServiceGet(discoveringServiceManager.getServiceURL(), "appWithServices");
+    String decodedResult = new Gson().fromJson(result, String.class);
```

Comment:
this line can be removed with 'result' being directly used.

Reply:
Do you mean reuse the result variable?

Reply:
yes, it looks like they have the same value.